### ISSUE TITLE
[java] Mockito verify method is not taken into account in JUnitTestsShouldIncludeAssert rule

### ISSUE THREAD

**Issue by GooDer** [pmd/pmd#358]
Please, prefix the report title with the language it applies to within brackets, such as *[java]* or *[apex]*. If not specific to a language, you can use *[core]*

**Rule Set:**
pmd-unit-tests:JUnitTestsShouldIncludeAssert

**Description:**
when unit test method includes only Mockito verify check then JUnit tests should include assert() or fail() is reported which is in this case false positive.

**Code Sample demonstrating the issue:**

```
    @Test
    public void doGet_ProvideNullFile_CheckResponse() throws ServletException, IOException {
      FileDownload download = new FileDownload();
      HttpServletRequest request = mock(HttpServletRequest.class);
      HttpServletResponse response = mock(HttpServletResponse.class);
      
      
      download.doGet(request, response);
      verify(response).sendError(eq(HttpServletResponse.SC_FORBIDDEN), anyString());
   }
```

**Running PMD through:** *[Other]* - SonarQube

**Comment by jsotuyod** [comment]
@GooDer thanks for the suggestion! I've the idea this was already discussed on the old sourceforge tracker, but I can't seem to find it.

Anyway, this sounds reasonable, we could probably add this is an enhancement.

**Comment by Zatara7** [comment]
Sorry to be commenting on closed ticket, but the check is should be smarter. It doesn't recognize if I do `BDDMockito.verify()`

**Comment by jsotuyod** [comment]
@Zatara7 we could probably just move the method whitelist to a property and let users customize it at will, or we will keep getting reports like this for `PowerMock.verifyStatic` and every other mocking library out there. I'm tracking this as #1084

**Comment by yasmimc** [comment]
Hi! I also found a case of false positive, when using verify() inside a foreach this warning is also triggered
